File: yejee/week10/의상.java
import java.util.Map;
import java.util.HashMap;

class Solution {
    public int solution(String[][] clothes) {
        int answer = 1;
        int count = 0;
        
        Map<String, Integer> clothesCnt = new HashMap<>();
        for(int i = 0 ; i < clothes.length ; i++){
            count = 0;
            if(clothesCnt.containsKey(clothes[i][1])){
                count = clothesCnt.get(clothes[i][1]);
            }
            clothesCnt.put(clothes[i][1], count + 1);
        }
        
        for(Map.Entry<String, Integer> entry : clothesCnt.entrySet()){
            answer *= entry.getValue() + 1;
        }
        answer -= 1;
        
        return answer;
    }
}

File: seoyeon/week10/의상.py
def solution(clothes):
    answer = 1
    dict = {}
    for i in clothes :
        if i[1] in dict.keys() :
            dict[i[1]] += 1
        else : dict[i[1]] = 1
    for i in dict :
        answer *= (dict[i] + 1)
    return answer - 1